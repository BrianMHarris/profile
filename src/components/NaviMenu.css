.navmenu-container {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  height: 100%;
  /*background-color: var(--clr-bg-navi);*/
}

.navmenu-btns {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
}

.navmenu-horzline {
  display: none;
}

.navmenu-contact-links {
  width: 72px;
}

/* Navbar Collapse Point */
@media screen and (max-width: 760px) {
  .navmenu-container {
    height: 100%;
    flex-direction: column;
    justify-content: space-evenly;
    align-items: space-between;
    background-color: var(--clr-bg-navi);
  }

  .navmenu-btns {
    height: 70%;
    flex-direction: column;
    justify-content: space-between;
    align-items: space-between;
  }

  .navmenu-horzline {
    display: initial;
    height: 1px;
    width: 75%;
    background-color: var(--clr-bg-main);
    margin: 5px 0;
  }

  .navmenu-contact-links {
    width: 130px;
  }
}
