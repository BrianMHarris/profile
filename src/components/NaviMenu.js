import React from 'react';
import './NaviMenu.css';
import Button from './Button'
import NaviContacts from '../components/NaviContacts';

// A section of menu "items" (buttons in this case).

const NaviMenu = (props) => (
  <div className="navmenu-container">
    <div className="navmenu-btns">
      <Button label="Home"
              addClasses="btn-hvr btn-nav"
              link="#home"
              handleClick={props.handleClick}/>
      <Button label="Work"
              addClasses="btn-hvr btn-nav"
              link="#projects"
              handleClick={props.handleClick}/>
      <Button label="About"
              addClasses="btn-hvr btn-nav"
              link="#about"
              handleClick={props.handleClick}/>
      <Button label="Contact"
              addClasses="btn-hvr btn-nav"
              link="#contact"
              handleClick={props.handleClick}/>
    </div>
    <div className="navmenu-horzline"></div>
    <div className={`navmenu-contact-links`}>
      <NaviContacts />
    </div>
  </div>
)

export default NaviMenu;
