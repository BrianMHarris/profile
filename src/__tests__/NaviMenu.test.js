import React from 'react';
import ReactDOM from 'react-dom';
import NaviMenu from '../components/NaviMenu';
import renderer from "react-test-renderer";

describe('Navi', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <NaviMenu />,
      div);
  });

  it('matches the snapshot', () => {
    const component = renderer.create(
        <NaviMenu />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
