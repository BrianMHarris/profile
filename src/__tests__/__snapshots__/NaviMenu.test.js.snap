// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navi matches the snapshot 1`] = `
<div
  className="navmenu-container"
>
  <div
    className="navmenu-btns"
  >
    <a
      className="btn-link"
      href="#home"
      onClick={[Function]}
    >
      <button
        className="btn-std btn-hvr btn-nav"
        onClick={null}
        onMouseOut={undefined}
        onMouseOver={undefined}
        type={undefined}
      >
        Home
      </button>
    </a>
    <a
      className="btn-link"
      href="#about"
      onClick={[Function]}
    >
      <button
        className="btn-std btn-hvr btn-nav"
        onClick={null}
        onMouseOut={undefined}
        onMouseOver={undefined}
        type={undefined}
      >
        About
      </button>
    </a>
    <a
      className="btn-link"
      href="#work"
      onClick={[Function]}
    >
      <button
        className="btn-std btn-hvr btn-nav"
        onClick={null}
        onMouseOut={undefined}
        onMouseOver={undefined}
        type={undefined}
      >
        Work
      </button>
    </a>
    <a
      className="btn-link"
      href="#projects"
      onClick={[Function]}
    >
      <button
        className="btn-std btn-hvr btn-nav"
        onClick={null}
        onMouseOut={undefined}
        onMouseOver={undefined}
        type={undefined}
      >
        Projects
      </button>
    </a>
    <a
      className="btn-link"
      href="#contact"
      onClick={[Function]}
    >
      <button
        className="btn-std btn-hvr btn-nav"
        onClick={null}
        onMouseOut={undefined}
        onMouseOver={undefined}
        type={undefined}
      >
        Contact
      </button>
    </a>
  </div>
  <div
    className="navmenu-horzline"
  />
  <div
    className="navmenu-contact-links"
  >
    <div>
      <div
        className="navcontacts-container"
      >
        <div
          className="ncitem-container"
        >
          <a
            href="https://www.linkedin.com/in/brian-harris-b875995"
          >
            <div
              className="ncitem-reveal"
            >
              <img
                alt="IMG"
                src="nav-linkedin-hover.svg"
              />
            </div>
            <div
              className="ncitem-overlay"
            >
              <img
                alt="IMG"
                src="nav-linkedin.svg"
              />
            </div>
          </a>
        </div>
        <div
          className="ncitem-container"
        >
          <a
            href="https://github.com/BrianMHarris"
          >
            <div
              className="ncitem-reveal"
            >
              <img
                alt="IMG"
                src="nav-github-hover.svg"
              />
            </div>
            <div
              className="ncitem-overlay"
            >
              <img
                alt="IMG"
                src="nav-github.svg"
              />
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
